utxo/src/utxo_impl/utxo_storage/test.rs

```diff
@@ -0,0 +1,356 @@
+use super::in_memory::UtxosDBInMemoryImpl;
+use super::*;
+use crate::test_helper::{convert_to_utxo, create_tx_inputs, create_tx_outputs, create_utxo};
+use crate::utxo_impl::{
+    flush_to_base, utxo_storage::UtxosDB, FlushableUtxoView, Utxo, UtxoEntry, UtxosCache, UtxosView,
+};
+use crate::ConsumedUtxoCache;
+use common::chain::block::timestamp::BlockTimestamp;
+use common::chain::config::create_mainnet;
+use common::chain::signature::inputsig::InputWitness;
+use common::chain::{Destination, OutPointSourceId, Transaction, TxInput, TxOutput};
+use common::primitives::{Amount, BlockHeight, Idable};
+use common::primitives::{Id, H256};
+use crypto::random::{make_pseudo_rng, seq, Rng};
+use itertools::Itertools;
+use std::collections::{BTreeMap, HashMap};
+
+fn create_transactions(
+    inputs: Vec<TxInput>,
+    max_num_of_outputs: usize,
+    num_of_txs: usize,
+) -> Vec<Transaction> {
+    // distribute the inputs on the number of the transactions specified.
+    let input_size = inputs.len() / num_of_txs;
+
+    // create the multiple transactions based on the inputs.
+    inputs
+        .chunks(input_size)
+        .into_iter()
+        .map(|inputs| {
+            let outputs = if max_num_of_outputs > 1 {
+                let rnd = make_pseudo_rng().gen_range(1..max_num_of_outputs);
```

Comment:
Maybe the new seedable RNG mechanism should be used here.

Reply:
Another TODO